RFE: add linters and enable GitHub CI

kolyshkin opened the pull request:
This PR
* fixes a few found linter warnings (all very minor)
* adds golangci-lint to makefile, replacing both golint (which is obsoleted) and govet (which is called by golangci-lint)
* enables some basic github CI checks for master branch and PRs:
  * golangci-lint
  * build and test checks for both supported golang versions (currently 1.16.x and 1.17.x) and the last unsupported one (1.15.x)

Fixes: #60

* v2 (Aug 19): added Go 1.17.x

kolyshkin commented:
Updated to add Go 1.17

kolyshkin commented:
@pcmoore @drakenclimber PTAL 🙏🏻 

This currently uses whatever libseccomp is available in Ubuntu-20.04. I can certainly add compile/test with libseccomp 2.5.2.

rata reviewed (approved):
As an outside contributor, this LGTM, thanks! I want to highlight that this should also close #67

pcmoore commented:
This looks okay to me, but we should hear from @drakenclimber, especially with respect to the GH Actions implementation.  Also, while not strictly necessary for this PR as it can be done later, we should probably go ahead and remove the Travis CI config while we are at it (if done in this PR, please do it as a separate patch).
```
Acked-by: Paul Moore <paul@paul-moore.com>
```

drakenclimber commented on seccomp.go:
```diff
@@ -45,7 +45,7 @@ func init() {
 	// which is necessary on older versions of libseccomp in order to work
 	// correctly.
-	GetAPI()
+	_, _ = GetAPI()
```
This patch is no longer needed due to commit https://github.com/seccomp/libseccomp-golang/commit/ef015460ecc91e963a187dd865453cacb6f31899

kolyshkin replied:
Rebased to remove this.

drakenclimber reviewed (approved):
Looks good to me.  I'll merge the patchset in except for patch https://github.com/seccomp/libseccomp-golang/pull/65/commits/c2b914ede632014f5c7fde74ec85df40071971c3 which is already effectively committed.

We need to update the CI badge on the front page.  I added Issue https://github.com/seccomp/libseccomp-golang/issues/68 to address this.

`Reviewed-by: Tom Hromatka <tom.hromatka@oracle.com>`

pcmoore commented:
Thanks @drakenclimber, @kolyshkin, and @rata!

kolyshkin commented:
> Also, while not strictly necessary for this PR as it can be done later, we should probably go ahead and remove the Travis CI config while we are at it (if done in this PR, please do it as a separate patch).

done

kolyshkin commented:
Pushed a new version, rebasing on current tip. The only changes are:
* removed commit c2b914e (obsoleted by ef01546)
* added a commit that removes `.travis.yml`

kolyshkin commented:
> I can certainly add compile/test with libseccomp 2.5.2.

Do you think it makes sense to build/test against the latest libseccomp release, or whatever Ubuntu is providing (2.5.1 at the moment) is fine?

pcmoore commented:
> Do you think it makes sense to build/test against the latest libseccomp release, or whatever Ubuntu is providing (2.5.1 at the moment) is fine?

In general I would rather see upstream test against upstream as opposed to whatever happens to be the default in the CI.

drakenclimber commented:
Merged.  CI passed - see [here](https://github.com/seccomp/libseccomp-golang/runs/3570811836).

Let me know if you see any issues.

kolyshkin commented:
> In general I would rather see upstream test against upstream as opposed to whatever happens to be the default in the CI.
 
OTOH testing against whatever Ubuntu ships with make sense, too. Let me try to implement a matrix.

pcmoore commented:
> > In general I would rather see upstream test against upstream as opposed to whatever happens to be the default in the CI.
> 
> OTOH testing against whatever Ubuntu ships with make sense, too. Let me try to implement a matrix.

Let's stick to just upstream for now, I don't want to go off into the weeds just yet with distro testing.
